api/units.go

```diff
@@ -101,15 +101,33 @@ func (ur *unitsResource) set(rw http.ResponseWriter, req *http.Request, item str
 		return
 	}
 
+	newUnit := false
 	if eu == nil {
 		if len(su.Options) == 0 {
 			err := errors.New("unit does not exist and options field empty")
 			sendError(rw, http.StatusConflict, err)
+			return
 		} else if err := ValidateOptions(su.Options); err != nil {
 			sendError(rw, http.StatusBadRequest, err)
+			return
 		} else {
-			ur.create(rw, su.Name, &su)
+			newUnit = true
 		}
+	} else if eu.Name == su.Name && len(su.Options) > 0 {
+		// There is already a unit with the same name
+		// check the hashes if they do not match then we will
+		// create a new unit with the same name and later
+		// the job will be updated to this new unit.
```

Comment:
Well, the commit messages and code changes *seem* to suggest that just submitting a unit with the same name will replace the existing one; which was my original understanding of how this is supposed to work. It's the code comment here that threw me off. I think the wording is *extremely* confusing -- especially the ominous "later"... I suspect what you are actually trying to say is that the reconciler will update the running jobs once the unit file has been replaced in the registry -- but that's not clear at all.